### PR TITLE
Add Netlify deployment scripts

### PR DESCRIPTION
**Pull request by matus-tomlein**
This PR shamelessly copies the scripts in the advanced analytics for Web accelerator to build the Hugo site using the template and also adds the CI action to deploy to Netlify.

I haven't tested the Netlify deployment and don't know much about Netlify, so hope that you can help me verify that.

### CODE REVIEW

**Comment by paulboocock** [comment]
I haven't quite nailed the deployment yet due to wanting to run the site as a rewrite on a path under docs.snowplow.io. 

I'll comment here when I've tweaked the Hugo build so that all the assets work on the rewritten netlify site.

**Comment by matus-tomlein** [comment]
@paulboocock I think it should be good now, please give it a look in case I missed something. Also opened a PR in the documentation to add the redirect: https://github.com/snowplow/documentation/pull/109

File: README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+```
+./scripts/build.sh
+```
+
+## Usage
+
+Run an HTTP server inside the `public` folder. For example, using Python:
+
+```
+python3 -m http.server --directory public
```
Comment (paulboocock) [inline README.md]: I'm a big fan of `npx http-server public` but maybe thats just the node in me speaking.

I wonder if we should pop in some `hugo server` docs?

Either way, the advanced analytics repo could do with this too!

**Review by paulboocock — CHANGES_REQUESTED** [review]
I think you need to run `git submodule update --init --remote` to update the submodule from `remote` to get the latest landing page changes (and to fix the logo [I hope]).

File: README.md (hunk 1 above)
Comment (matus-tomlein) [reply README.md]: I couldn't resist a bit of shell scripting and updated the build script to be able to start a Hugo server directly using:

```
./scripts/build.sh serve
```

If it's okay, we can also add this to the web accelerator.

Comment (paulboocock) [reply README.md]: Nice! Would be great to see this on the web one too.

If I'm being super nitpicky, it should probably be `run.sh` or something now, `build.sh build` seems weird.

**Review by paulboocock — APPROVED** [review]
LGTM.

If you wish, you can remove `.github/workflows/deploy.yml` now, pretty sure it won't work on the same URLs anyway after this goes live. I sort of gave upo trying to get to work on github and netlify (with rewrites) at the same time and maintain the urls.

Jack also pushed another update to the template today, which makes the "Introduction" link bold when active. Could update the submodule again if you wish.

And lastly, there's still a demo folder in the root of this repo. Its doing no harm but I'm sure if you need it. I guess you have plans but just a reminder incase you've changed them :-)

**Pull request merged**